internal/pkg/cli/init.go

```diff
@@ -286,6 +282,66 @@ func newInitOpts(vars initVars) (*initOpts, error) {
 				o.initWlCmd = &opts
 				o.schedule = &opts.schedule // Surfaced via pointer for logging
 				o.initWkldVars = &opts.initWkldVars
+			case t == manifestinfo.StaticSiteType:
+				ws, err := workspace.Use(fs)
+				if err != nil {
+					var (
+						errWorkspaceNotFound *workspace.ErrWorkspaceNotFound
+					)
+					if !errors.As(err, &errWorkspaceNotFound) {
+						return err
+					}
+					sourceFiles, err := selector.AskCustomPaths(o.prompt, staticSiteInitDirFilePathPrompt,
+						staticSiteInitDirFilePathHelpPrompt, func(v interface{}) error {
+							return validatePath(fs, v)
+						})
+					if err != nil {
+						return fmt.Errorf("ask for paths to source files: %w", err)
+					}
+					wkldVars.sourcePaths = sourceFiles
+				}
+				sel, err := selector.NewLocalFileSelector(prompt, fs, ws)
+				if err != nil {
+					return err
+				}
+				svcVars := initSvcVars{
+					initWkldVars: wkldVars,
+					port:         vars.port,
+					ingressType:  ingressTypeInternet,
+				}
+				opts := initSvcOpts{
+					initSvcVars: svcVars,
+
+					fs:                fs,
+					sel:               dfSel,
+					sourceSel:         sel,
+					store:             configStore,
+					topicSel:          snsSel,
+					prompt:            prompt,
+					dockerEngine:      dockerengine.New(cmd),
+					wsPendingCreation: true,
```

Comment:
Ahh! Actually, this reminds me that "static assets"` is actually not the only service that faces this chicken-egg dilemma! Lookie here: https://github.com/aws/copilot-cli/blob/222ea9cf4fab097abd7229380d661ba1a5f66fae/internal/pkg/cli/job_init.go#L146

I wonder if we can do something similar for static asset:
```go
// In cli/init.go
setupWorkloadInit: func(o *initOpts, wkldType string) error {
   case manifestinfo.IsTypeAService(t):
      var wsPendingCreation bool
      ws, err := workspace.Use(fs)
      var errWorkspaceNotFound *workspace.ErrWorkspaceNotFound
      if err != nil {
        var errWorkspaceNotFound *workspace.ErrWorkspaceNotFound
        if !errors.As(err, &errWorkspaceNotFound) {
            return err
         }
        wsPendingCreation = true
      } 
      opts := initSvcOpts{		
        // ....other stuff is the same
        wsPendingCreation: true,
      }
}


// In cli/svc_init.go
func (o *initSvcOpts) askStaticSite() error {
        if o.wsPendingCreation {
            return AskCustomPath(...)
        }
	var assets []manifest.FileUpload
	sources, err := o.askSource()
	if err != nil {
		return err
	}
	for _, source := range sources {
		info, err := o.fs.Stat(source)
		if err != nil {
			return fmt.Errorf("get info for %q: %w", source, err)
		}
		assets = append(assets, manifest.FileUpload{
			Source:    source,
			Recursive: info.IsDir(),
		})
	}
	o.staticAssets = assets
	return nil
}
```

My main concern with the current approach is that we are prompting questions in `setupWorkloadInit`, which was meant to set up workload init commands. We were supposed to ask questions only in `Ask`, and if we break this diagram, it might be come difficult to debug/maintain the code.

Reply:
Yeah, that totally makes sense! This is much more streamlined and organized.

Reply:
We never actually check for a workspace before setting `wsPendingCreation` to true, so if someone is just adding a workload using `copilot init`, we act like it's not there. I'll add checks.